tests/filecheck/transforms/csl-stencil-handle-async-flow.mlir

```diff
@@ -0,0 +1,174 @@
+// CHECK-NEXT:       ^2(%arg5 : memref<4x255xf32>, %arg6 : index, %arg7 : memref<510xf32>):
+// CHECK-NEXT:         %47 = csl_stencil.access %arg5[1, 0] : memref<4x255xf32>
+// CHECK-NEXT:         %48 = csl_stencil.access %arg5[-1, 0] : memref<4x255xf32>
+// CHECK-NEXT:         %49 = csl_stencil.access %arg5[0, 1] : memref<4x255xf32>
+// CHECK-NEXT:         %50 = csl_stencil.access %arg5[0, -1] : memref<4x255xf32>
+// CHECK-NEXT:         %51 = memref.subview %arg7[%arg6] [255] [1] : memref<510xf32> to memref<255xf32, strided<[1], offset: ?>>
+// CHECK-NEXT:         "csl.fadds"(%51, %50, %49) : (memref<255xf32, strided<[1], offset: ?>>, memref<255xf32>, memref<255xf32>) -> ()
+// CHECK-NEXT:         "csl.fadds"(%51, %51, %48) : (memref<255xf32, strided<[1], offset: ?>>, memref<255xf32, strided<[1], offset: ?>>, memref<255xf32>) -> ()
+// CHECK-NEXT:         "csl.fadds"(%51, %51, %47) : (memref<255xf32, strided<[1], offset: ?>>, memref<255xf32, strided<[1], offset: ?>>, memref<255xf32>) -> ()
+// CHECK-NEXT:         csl_stencil.yield %arg7 : memref<510xf32>
+// CHECK-NEXT:       }, {
+// CHECK-NEXT:       ^3(%arg5_1 : memref<512xf32>, %arg6_1 : memref<510xf32>):
+// CHECK-NEXT:         %52 = memref.subview %arg5_1[2] [510] [1] : memref<512xf32> to memref<510xf32, strided<[1], offset: 2>>
+// CHECK-NEXT:         %53 = memref.subview %arg5_1[0] [510] [1] : memref<512xf32> to memref<510xf32, strided<[1]>>
+// CHECK-NEXT:         "csl.fadds"(%arg6_1, %arg6_1, %53) : (memref<510xf32>, memref<510xf32>, memref<510xf32, strided<[1]>>) -> ()
+// CHECK-NEXT:         "csl.fadds"(%arg6_1, %arg6_1, %52) : (memref<510xf32>, memref<510xf32>, memref<510xf32, strided<[1], offset: 2>>) -> ()
+// CHECK-NEXT:         %54 = arith.constant 1.666600e-01 : f32
+// CHECK-NEXT:         "csl.fmuls"(%arg6_1, %arg6_1, %54) : (memref<510xf32>, memref<510xf32>, f32) -> ()
+// CHECK-NEXT:         "csl.call"() <{"callee" = @for_inc0}> : () -> ()
+// CHECK-NEXT:         csl_stencil.yield %arg6_1 : memref<510xf32>
+// CHECK-NEXT:       }) to <[0, 0], [1, 1]>
+// CHECK-NEXT:       csl.return
+// CHECK-NEXT:     }
+// CHECK-NEXT:     csl.func @for_inc0() {
+// CHECK-NEXT:       %47 = arith.constant 1 : i16
+// CHECK-NEXT:       %48 = "csl.load_var"(%38) : (!csl.var<i16>) -> i16
+// CHECK-NEXT:       %49 = arith.addi %48, %47 : i16
+// CHECK-NEXT:       "csl.store_var"(%38, %49) : (!csl.var<i16>, i16) -> ()
+// CHECK-NEXT:       %50 = "csl.load_var"(%39) : (!csl.var<memref<512xf32>>) -> memref<512xf32>
+// CHECK-NEXT:       %51 = "csl.load_var"(%40) : (!csl.var<memref<512xf32>>) -> memref<512xf32>
+// CHECK-NEXT:       "csl.store_var"(%39, %51) : (!csl.var<memref<512xf32>>, memref<512xf32>) -> ()
+// CHECK-NEXT:       "csl.store_var"(%40, %50) : (!csl.var<memref<512xf32>>, memref<512xf32>) -> ()
```

Comment:
This sequence of load then stores appears to be unnecessary?

Reply:
It's doing a flip, as the scf.yield is swapping the arguments. In this example, we load them and store them in the order prescribed, ie. reversed.

Reply:
Added a comment.